### ISSUE TITLE
MobX: tools / data count doesn't exist in MobX yet

### ISSUE THREAD

**Issue by AnaBelgun** [TerriaJS/terriajs#4389]
In a non-MobX map you can use the url/#tools=1 to count the nr of datasets and groups in the catalogue; this currently doesn't exist in MobX branch yet

![image](https://user-images.githubusercontent.com/18452161/83365361-ce7a0780-a3ea-11ea-8958-9f815758ea6f.png)

**Comment by soyarsauce** [comment]
it will need adaptation for magda based maps or any map really that uses magdareferences cause those are loaded on demand, don't know how many datasets/groups there are unless you make terriajs-mobx force load all the groups

**Comment by zoran995** [comment]
Hi @AnaBelgun, is this something we still want to add to v8 or can it be considered a deprecated/removed feature, as v8 was released quite some time ago?

**Comment by AnaBelgun** [comment]
Yes, we still want it. I believe we even have a draft pt for it. It's very
useful.

**Comment by zoran995** [comment]
From #2991 

> It would be really useful if the count tool was hierarchical with the possibility of digging down the see counts for subtrees. Something like:
> 
> ```
> The catalog contains 674 items in 182 groups.
> Electricity Infrastructure: 57 items / 15 groups
> - Generation: 12 items / 5 groups
> -- Current Power Generation: 6 items / 0 groups
> -- [...]
> - Transmission: 5 items / 3 groups
> [...]
> Renewable Energy: 415 items / 128 groups
> [...]
> ```

and #2713 

> I used the tools to count datasets and I get different counts.
> The first time I went to the site and started the Count straight away, on another tab I went to the site, opened the catalogue and searched for something, and only once the Loading finished did I exit and start the Count. The second count got significantly higher number of datasets.
